kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriberServiceImpl.java

```diff
@@ -33,7 +43,50 @@ public void add(Subscriber subscriber) {
 
     @Override
     public void update(Subscriber subscriber) {
+
+        // cache the previous version of the subscriber in order to compare before/after
+        Subscriber retrievedSubscriber = subscriberDataService.findByCallingNumber(subscriber.getCallingNumber());
+
         subscriberDataService.update(subscriber);
+
+        // update start dates for subscriptions if reference date (LMP/DOB) has changed
+        Set<Subscription> subscriptions = retrievedSubscriber.getActiveSubscriptions();
+        Iterator<Subscription> subscriptionIterator = subscriptions.iterator();
+        Subscription subscription;
+
+        while (subscriptionIterator.hasNext()) {
+            subscription = subscriptionIterator.next();
+
+            if ((subscription.getSubscriptionPack().getType() == SubscriptionPackType.PREGNANCY) &&
+                    (retrievedSubscriber.getLastMenstrualPeriod() != subscriber.getLastMenstrualPeriod()))  {
+
+                updateOrSubscribe(subscriber, subscription, subscriber.getLastMenstrualPeriod(),
```

Comment:
Should this use subscriber.getLastMenstrualPeriod or retrievedSubscriber.getLastMenstrualPeriod()?